### DISCUSSION TITLE
Remove default packages

### DISCUSSION THREAD

**Discussion by bukk530** [eclipse-theia/theia#8442] [General] [answered]
Hi all.

I need to build a simple ide that only includes the monaco editor, tabs, and the file navigator. To do so I've removed unnecessary packages from the `packages `folder, but for every package I remove it seems to break the compilation with some error. 

Is there any recommended way to get a minimal editor without all the features that theia includes, and possibly getting rid of the backend and running it soley on the browser?

Thanks.

**Comment by vince-fugnitto** [comment, marked as answer]
@bukk530 thank you for the question, you should not modify this repository in hopes or building your own application. Instead, you should compose your own applications with the extensions you need and remove any functionality which you do not require with custom extensions and dependency injection.

Please review the following documentation related to building your own application:
- https://theia-ide.org/docs/composing_applications/

It is not possible to run without a backend, but both the `browser` and `electron` (desktop) targets are supported.
